### PR TITLE
Add conda GPU example

### PR DESCRIPTION
**Pull request by agitter**
The adapts the existing PyTorch Docker example to run PyTorch inside a GPU-enabled conda environment.

### CODE REVIEW

File: conda/pytorch_cnn.sub [hunk 1]
```diff
@@ -0,0 +1,25 @@
+universe = vanilla
+executable = pytorch_cnn.sh
+output = $(Cluster)_$(Process).out
+
+transfer_input_files = environment.yml, ../docker/pytorch_python/main.py, ../docker/pytorch_python/MNIST_data.tar.gz
```
Comment (ChristinaLK) [inline conda/pytorch_cnn.sub:5]: I have mixed feelings about this organization - would it make sense to have the data in a standalone folder?

File: conda/pytorch_cnn.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+sh miniconda.sh -b -p $HOME/miniconda3
+rm miniconda.sh
+export PATH=$HOME/miniconda3/bin:$PATH
+
+# Update conda as workaround for https://github.com/conda/conda/issues/9681
+# Will no longer be needed once conda >= 4.8.3 is available from repo.anaconda.com
+conda install conda=4.8.3
+
+# Set up conda
+source $HOME/miniconda3/etc/profile.d/conda.sh
```
Comment (ChristinaLK) [inline conda/pytorch_cnn.sh:26]: does this work? I've had `source` fail on non-interactive jobs...

File: conda/pytorch_cnn.sub (hunk 1 above)
Comment (agitter) [reply conda/pytorch_cnn.sub:5]: I agree it's clunky.  Should we make a top-level `resources` or `shared` directory?

File: conda/pytorch_cnn.sh (hunk 2 above)
Comment (agitter) [reply conda/pytorch_cnn.sh:26]: It worked on the batch jobs I submitted to test this.  I'd be curious to learn under which conditions it fails but expect that is hard to determine.

What fails?  The `source` command doesn't exist?

Comment (ChristinaLK) [reply conda/pytorch_cnn.sh:26]: yes, the `source` command doesn't exist.

File: conda/pytorch_cnn.sub (hunk 1 above)
Comment (ChristinaLK) [reply conda/pytorch_cnn.sub:5]: I think a `shared` directory makes sense.

Comment (lmichael107) [reply conda/pytorch_cnn.sub:5]: +1

File: conda/pytorch_cnn.sh (hunk 2 above)
Comment (lmichael107) [reply conda/pytorch_cnn.sh:26]: @ChristinaLK can you describe more? 
There's differential behavior (`source` availability) in CHTC? (If so, then we should address it, but keep in mind Flocking/Glidein implications.) 
Is it a difference in shell language (e.g. bash vs. sh)? 
Can we get around it with `getenv = true`?

File: conda/pytorch_cnn.sub (hunk 1 above)
Comment (agitter) [reply conda/pytorch_cnn.sub]: Updated to use `shared` directory in 61f2fc0a05b7eec4363b014c2ea5d891934afc7e  Is this what you had in mind?

I confirmed the conda PyTorch example still runs after these changes.  The Docker PyTorch example is in the queue.

Comment (ChristinaLK) [reply conda/pytorch_cnn.sub]: looks good!

File: conda/pytorch_cnn.sh (hunk 2 above)
Comment (ChristinaLK) [reply conda/pytorch_cnn.sh:26]: I'm happy to let this be as is and we can fix later.

Comment (agitter) [reply conda/pytorch_cnn.sh:26]: Sounds good to me.  If `source` causes an issue later, I'm willing to rewrite this to use a workaround.

**Comment by agitter** [comment]
Can we please merge this pull request if everything looks good?  I may follow up with another pull request that will also use the shared PyTorch example files.

**Pull request merged**